Sprint 3

VictorMugisha opened the pull request:
## Tasks Completed
1. [(Review) Add GitHub Actions Workflow for ESLint and Prettier Checks](https://trello.com/c/C4OrcnkW/144-review-add-github-actions-workflow-for-eslint-and-prettier-checks)
2. [(Admin) View form: Hide "Add Stage" button](https://trello.com/c/ss7JUcs2/137-admin-view-form-hide-add-stage-button)
3. [(Admin) Create users' (Prospect, coach) edit modal](https://trello.com/c/LOEKaxol/142-admin-create-users-prospect-coach-edit-modal)
4. [(Admin) Edit coach on coach overview](https://trello.com/c/450rdi9d/101-admin-edit-coach-on-coach-overview)

KevinRebakure committed:
refactor: organise imports

KevinRebakure committed:
feat: log a message when disconnecting a socket

AL2002MI08 committed:
refactor: isSignup logic

InshutiSouvede committed:
fix: hide add stage button when the form is opened in view mode

AL2002MI08 committed:
feat: add alert for successfully verifying email

InshutiSouvede committed:
feat: create a modal for editing a user

InshutiSouvede committed:
chore: update message on successful user modification

dependabot[bot] committed:
chore(deps-dev): bump vite from 6.2.3 to 6.2.4 (#106)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.3 to 6.2.4.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.4/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.4/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

InshutiSouvede committed:
Merge branch 'dev' into ft/update-user-modal

InshutiSouvede committed:
Merge 'dev' into fix/add-stage-button

AL2002MI08 committed:
feat: add edit icon on coaches overview

InshutiSouvede committed:
chore: resolve format issues

InshutiSouvede committed:
Merge pull request #100 from kb21-rw/fix/add-stage-button

(Admin) View form: Hide "Add Stage" button.

InshutiSouvede committed:
chore: resolve format issues

AL2002MI08 committed:
feat: implement edit coach functionality by admin

AL2002MI08 committed:
Merge 'dev' into ft/edit-coach-overview

KevinRebakure committed:
fix: search input field

AL2002MI08 committed:
refactor: alert message of updating coach information

AL2002MI08 committed:
refactor: aria description for editing coach details

AL2002MI08 committed:
Merge pull request #102 from kb21-rw/fix/invalid-access-token

(Prospects) Email Verification and Login: Incorrect Alert Messages Displayed.

AL2002MI08 committed:
refactor: state management of coach details

AL2002MI08 committed:
fix: prettier format in global layout

AL2002MI08 committed:
chore: add line spacing

AL2002MI08 committed:
Merge 'dev' into ft/edit-coach-overview

AL2002MI08 committed:
chore: merge changes from dev

AL2002MI08 committed:
Merge pull request #113 from kb21-rw/ft/edit-coach-overview

(Admin) Edit coach on coach overview

AL2002MI08 committed:
Merge pull request #109 from kb21-rw/ft/update-user-modal

(Admin) create user's edit modal

KevinRebakure committed:
Merge pull request #110 from kb21-rw/fix/searchfields-deleting

(Admin) Search Field Functionality Issues on all forms page

musayann commented on src/components/layouts/GlobalLayout.tsx:
```diff
@@ -45,11 +58,44 @@ export default function GlobalLayout() {
     }
-  }, [user, userError, dispatch, isLoading])
+
+    if (alertData) {
```
I would suggest to define functions outside the useEffect hook and just trigger them in the useEffect

musayann replied:
Avoid adding complex logic and statements inside the useEffect

dependabot[bot] committed:
chore(deps-dev): bump vite from 6.2.4 to 6.2.5 (#116)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 6.2.4 to 6.2.5.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.5/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.5/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 6.2.5
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

AL2002MI08 committed:
refactor: move alert data functionality definition outside of useEffect

AL2002MI08 committed:
refactor: remove unnecessary conditions for error handling

AL2002MI08 committed:
Merge pull request #119 from kb21-rw/fix/alert-data-functionality

Refactor alert logic to prevent redefinition
